build-tools/automation/yaml-templates/run-msbuild-device-tests.yaml

```diff
@@ -0,0 +1,76 @@
+# Runs MSBuild tests against a device running on macOS
+
+parameters:
+  job_name: ''
+  job_suffix: ''
+  nunit_categories: ''
+  target_framework: 'net472'
+
+jobs:
+  - job: ${{ parameters.job_name }}
+    displayName: MSBuild With Emulator - macOS - ${{ parameters.job_suffix }}
+    pool: $(HostedMac)
+    timeoutInMinutes: 150
+    cancelTimeoutInMinutes: 5
+    workspace:
+      clean: all
+    variables:
+      UseDotNet: ${{ eq(parameters.target_framework, 'netcoreapp3.1') }}
+    steps:
+    - template: setup-test-environment.yaml
+
+    - script: mono $(System.DefaultWorkingDirectory)/build-tools/xaprepare/xaprepare/bin/$(XA.Build.Configuration)/xaprepare.exe --s=EmulatorTestDependencies --no-emoji --run-mode=CI
+      displayName: install emulator
+
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        artifactName: $(TestAssembliesArtifactName)
+        downloadPath: $(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)
+
+    - task: MSBuild@1
+      displayName: build check-boot-times.csproj
+      inputs:
+        solution: build-tools/check-boot-times/check-boot-times.csproj
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /restore /bl:$(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/build-check-boot-times.binlog
+      continueOnError: true
+
+    - task: MSBuild@1
+      displayName: Run check-boot-times
+      inputs:
+        solution: tests/Mono.Android-Tests/Mono.Android-Tests.csproj
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:CheckBootTimes /bl:$(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/run-check-boot-times.binlog
+      continueOnError: true
+
+    - task: MSBuild@1
+      displayName: start emulator
+      inputs:
+        solution: tests/Mono.Android-Tests/Mono.Android-Tests.csproj
+        configuration: $(XA.Build.Configuration)
+        msbuildArguments: /t:AcquireAndroidTarget /bl:$(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/start-emulator.binlog
```

Comment:
@pjcollins you might double-check that I fixed the merge conflict and didn't lose anything.

Do we need to run this twice? Should I put a condition so it only runs for "Legacy" and not .NET 6?

Reply:
This looks good to me, and yeah we should make this conditional to run only once like you said:

```yaml
    - ${{ if eq(parameters.job_suffix, 'Legacy') }}:
      - task: MSBuild@1
        displayName: build check-boot-times.csproj
        inputs:
          solution: build-tools/check-boot-times/check-boot-times.csproj
          configuration: $(XA.Build.Configuration)
          msbuildArguments: /restore /bl:$(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/build-check-boot-times.binlog
        continueOnError: true

      - task: MSBuild@1
        displayName: Run check-boot-times
        inputs:
          solution: tests/Mono.Android-Tests/Mono.Android-Tests.csproj
          configuration: $(XA.Build.Configuration)
          msbuildArguments: /t:CheckBootTimes /bl:$(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/run-check-boot-times.binlog
        continueOnError: true
```